trackingValidation.py

```diff
@@ -180,32 +282,44 @@ def trackReleaseValidation(filename1, filename2, tag):
+                histos_nested[label][coll].update({name: dqm_files[label][coll][name].to_hist()})
 
     for coll in trackCollections.values():
         makedir(os.path.join(plotter.savedir, coll))
         for name, labels in names_nested.items():
-            plotter.plotHistos(h1=histos_nested["pre3"][coll][name],
-                               h2=histos_nested["pre4"][coll][name],
-                               label1="pre3",
-                               label2="pre4",
+            v_histos = [histos_nested[label][coll][name] for label in dqm_paths.keys()]
+            v_labels = [label for label in dqm_paths.keys()]
+            plotter.plotHistos(v_h=v_histos,
+                               v_label=v_labels,
                                ylabel=labels[0],
                                xlabel=labels[1],
                                title=coll,
                                savename=coll + '/' + name)
             
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Make track validation plots.')
-    parser.add_argument('--filename1', type=str, required=True, help='Name of the first ROOT filename.')
-    parser.add_argument('--filename2', type=str, required=True, help='Name of the second ROOT filename.')
-    parser.add_argument('--tag', type=str, required=True, help='Tag to uniquely identify the plots.')
+    parser.add_argument('--files', type=str, required=True, help='Comma-separated list of paths to the ROOT files.')
+    parser.add_argument('--labels', type=str, required=False, help='Comma-separated list of labels for the legend.')
+    parser.add_argument('--tag', type=str, default=None, required=True, help='Tag to uniquely identify the plots.')
+    parser.add_argument('--odir', type=str, required=False, help='Path to the output directory (if not specified, save to current directory).')
     # parser.add_argument('--year', type=str, required=True,
     #                     choices=['2016', '2016APV', '2017', '2018'], help='Year')
     # parser.add_argument('--rebin', type=int, required=False, help="Rebin factor, leading to less bins.", default=1)
     # parser.add_argument('--pu', action='store_true', help='Using PU sample.')
     args = parser.parse_args()
 
-    trackReleaseValidation(filename1=args.filename1, filename2=args.filename2, tag=args.tag)
+    if ',' in args.files:       files = args.files.split(',')
+    else:                       files = [args.files]
+    if args.labels:
+        if ',' in args.labels:  labels = args.labels.split(',')
+        else:                   labels = [args.labels]
+    else:
+        labels = [f"File_{i}" for i in range(len(files))]
+    
+    assert len(files) == len(labels), "Number of files and labels must match."
+
+    trackReleaseValidation(files=files, labels=labels, tag=args.tag, odir=args.odir)
+    hgcalReleaseValidation(files=files, labels=labels, tag=args.tag, odir=args.odir)
```

Comment:
You can create a dict with the arguments and then pass the dict to the two functions, since the arguments are identical

```python
valid_args = dict(...)
trackReleaseValidation(**valid_args)
hgcalReleaseValidation(**valid_args)
```